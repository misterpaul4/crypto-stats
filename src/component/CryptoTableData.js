/* eslint-disable react/prop-types */
import React from 'react';
import { Link } from 'react-router-dom';
import { to2Decimal, toDecimal, moneyWithCommas } from '../utils';

const Crypto = ({
  crypto,
}) => {
  const percentTD = value => (
    <td className={value > '0' ? 'text-success' : 'text-danger'}>
      {to2Decimal(value)}
      %
    </td>
  );

  return (
    <tr className="chart-crypto-row">
      <td>{crypto.market_cap_rank}</td>
      <td>
        <Link to={`/cryptostat/${crypto.id}`}>
          <img src={crypto.image} alt="icon" className="mr-2" />
          {crypto.name}
        </Link>
      </td>
      <td>
        $
        {moneyWithCommas(to2Decimal(crypto.current_price))}
      </td>

      {percentTD(crypto.price_change_percentage_24h)}

      {percentTD(crypto.price_change_percentage_7d_in_currency)}

      <td>{moneyWithCommas(toDecimal(crypto.circulating_supply))}</td>

      <td>{crypto.max_supply ? moneyWithCommas(toDecimal(crypto.max_supply)) : '-'}</td>
      <td>
        $
        {moneyWithCommas(crypto.market_cap)}
      </td>
    </tr>
  );
};
export default Crypto;
