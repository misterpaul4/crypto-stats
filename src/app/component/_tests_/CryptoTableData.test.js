import ShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';
import CryptoTableData from '../CryptoTableData';

it('should render Table Data correctly without errors', () => {
  const crypto = {
    market_cap_rank: 1,
    image: 'url',
    name: 'coin',
    current_price: 54,
    price_change_percentage_24h: 58,
    price_change_percentage_7d_in_currency: 60,
    circulating_supply: 50000,
    max_supply: 857,
    market_cap: 87885,
  };

  const render = new ShallowRenderer();

  render.render(<CryptoTableData crypto={crypto} />);
  expect(render.getRenderOutput()).toMatchSnapshot();
});
