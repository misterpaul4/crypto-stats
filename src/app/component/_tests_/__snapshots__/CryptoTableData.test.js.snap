// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Table Data correctly without errors 1`] = `
<tr
  className="chart-crypto-row"
>
  <td>
    1
  </td>
  <td>
    <Link
      to="/cryptostat/undefined"
    >
      <img
        alt="icon"
        className="mr-2"
        src="url"
      />
      coin
    </Link>
  </td>
  <td>
    $
    54.00
  </td>
  <td
    className="text-success"
  >
    58.00
    %
  </td>
  <td
    className="text-success"
  >
    60.00
    %
  </td>
  <td>
    50,000
  </td>
  <td>
    857
  </td>
  <td>
    $
    87,885
  </td>
</tr>
`;
